Makefile

```diff
@@ -114,6 +116,7 @@ ifeq ($(IS_MINGW),1)
 EXT = dll
 AR_EXT = lib
 BIN_EXT = .exe
+UNICORN_CFLAGS := $(UNICORN_CFLAGS:-fPIC=)
```

Comment:
so this fixes MINGW compilation as well, not only CYGWIN?

can you please separate the pull request so it is easier to track down the fix?